src/category_theory/sites/types.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import category_theory.sites.canonical
+import category_theory.sites.sheaf
+
+/-!
+# Grothendieck Topology and Sheaves on the Category of Types
+
+In this file we define a grothendieck topology on the category of types,
+and construct the canonical function that sends a type to a sheaf over
+the category of types.
```

Comment:
Do you actually construct this function? I think a better description would be to say something like the yoneda presheaf for any type is a sheaf (or to actually make a function which takes types to sheaves).

Reply:
@kckennylau :arrow_up: what do you think?

Reply:
Now I have actually constructed the functor and the induced equivalence of categories.